### PR TITLE
[logger] log version_information events

### PR DESCRIPTION
**Pull request by pyfisch**
Use more events from qlog draft for easier debugging.

### CODE REVIEW

**Comment by jlaine** [comment]
This looks great, would you have a pointer to the corresponding specs?

**Comment by pyfisch** [comment]
> It looks as though there are several other cases where we should be logging this event?

Not sure, which cases do you think are missing?

File: src/aioquic/quic/connection.py [hunk 1]
```diff
@@ -789,7 +789,18 @@ def receive_datagram(self, data: bytes, addr: NetworkAddress, now: float) -> Non
...
-                if not common:
+                chosen_version = common[0] if common else None
+                if self._quic_logger is not None:
+                    self._quic_logger.log_event(
+                        category="transport",
+                        event="version_information",
+                        data={
+                            "server_versions": versions,
+                            "client_versions": self._configuration.supported_versions,
+                            "chosen_version": chosen_version,
```
Comment (jlaine) [inline src/aioquic/quic/connection.py:800]: I don't think passing `None` here is valid.

**Comment by jlaine** [comment]
> > It looks as though there are several other cases where we should be logging this event?
> 
> Not sure, which cases do you think are missing?

It looks like we should have `server_versions` ?

**Comment by jlaine** [comment]
Let's merge this as-is, we can always iterate.

**Pull request merged**